app/src/sharedTest/java/org/oppia/app/testing/ImageRegionSelectionTestActivityTest.kt

```diff
@@ -0,0 +1,53 @@
+package org.oppia.app.testing
+
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.action.GeneralClickAction
+import androidx.test.espresso.action.Press
+import androidx.test.espresso.action.Tap
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+
+@RunWith(AndroidJUnit4::class)
+class ImageRegionSelectionTestActivityTest {
+
+  @Test
+  fun testImageRegionSelectionTestActivity_clickonRegion3() {
+    launch(ImageRegionSelectionTestActivity::class.java).use { scenario ->
+      onView(withId(R.id.clickable_image_view)).perform(
+        clickAtXY(0.3f, 0.3f)
+      )
+      onView(withId(R.id.selected_region_view)).check(
+        matches(isDisplayed())
+      )
+    }
+  }
+
+  private fun clickAtXY(pctX: Float, pctY: Float): ViewAction {
```

Comment:
I'm wondering if its worth keeping these kind of functions in a separate file so that it can be used in multiple tests, what do you think?

Reply:
Originally i did keep this in a seperate file but then i saw instances in which this kind kf function were in same file so that's why i moved it to test file

Reply:
Now that we have the testing module , we can leverage that to keep common test methods . @BenHenning what do you think?

Reply:
I think the idea of promoting reuse for Espresso actions--let's move this to its own file (preferably as part of the testing module) if possible.

Reply:
@aggarwalpulkit596 can you move this to its own file in the testing module?

Reply:
Not sure what has happened on this front , I cannot see this function anymore so I guess its not required?

Reply:
This is part of its own class now @vinitamurthi: ``ClickActions``.